### PR TITLE
Add history method

### LINKED ISSUE #24: Missing `history` operation
I cannot seem to find the `history` operation (https://www.hl7.org/fhir/http.html#history). Is the operation not yet implemented?

### PR DESCRIPTION
**Pull request by skomski**
Closes #24 
Update deps

#19 was a bit too much work right now :(

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -49,7 +49,7 @@ async fn main() -> Result<(), Error> {
     let settings = RequestSettings::default()
         .header(header::AUTHORIZATION, "Bearer <token>".parse().unwrap());
     let client = Client::builder()
-        .base_url("http://localhost:8090/fhir/".parse().unwrap())
+        .base_url("http://localhost:8100/fhir/".parse().unwrap())
```
Comment (FlixCoder) [inline README.md:52]: why is this changed?

File: crates/fhir-sdk/Cargo.toml [hunk 2]
```diff
@@ -36,22 +36,22 @@ r5 = ["fhir-model/r5"]
...
-async-trait = {version = "0.1.68", optional = true}
+async-trait = {version = "0.1.79", optional = true}
 fhir-model = {path = "../fhir-model", version = "0.9.0", default-features = false}
-futures = {version = "0.3.28", optional = true}
-reqwest = {version = "0.11.16", features = ["json"], optional = true}
-serde = {version = "1.0.159", optional = true}
-serde_json = {version = "1.0.95", optional = true}
-thiserror = {version = "1.0.40", optional = true}
+futures = {version = "0.3.30", optional = true}
+reqwest = {version = "0.12.2", features = ["json"], optional = true}
```
Comment (FlixCoder) [inline crates/fhir-sdk/Cargo.toml:42]: I think reqwest 0.12 needs the h2 feature as well enabled to support that right? Or is it enabled by default?

File: crates/fhir-sdk/tests/client-r4b.rs [hunk 3]
```diff
@@ -23,13 +23,71 @@ use fhir_sdk::{
...
+use reqwest::header::HeaderValue;
+use serde_json::json;
+use tokio::sync::OnceCell;
+
+fn extract_json_field(json_body: serde_json::Value, field: &str) -> Option<String> {
+	let code = json_body.as_object()?.get(field)?.as_str()?;
+	Some(code.to_owned())
+}
+
+async fn medplum_auth() -> Result<HeaderValue> {
```
Comment (FlixCoder) [inline crates/fhir-sdk/tests/client-r4b.rs:35]: R4B should be tested against an R4B server, so it needs to stay the HAPI one right?

**Review by FlixCoder — COMMENTED** [review]
Just a first look from my phone as I am on the way :D Need to take a closer look and test

File: crates/fhir-sdk/Cargo.toml (hunk 2 above)
Comment (skomski) [reply crates/fhir-sdk/Cargo.toml:42]: Enabled by default

File: crates/fhir-sdk/tests/client-r4b.rs (hunk 3 above)
Comment (skomski) [reply crates/fhir-sdk/tests/client-r4b.rs:35]: Yes forgot to remove it.
